engine/client/pathutil.go

```diff
@@ -56,3 +57,23 @@ func getDrive(path string) string {
 
 	return ""
 }
+
+// expandPath expands a given path to its absolute form, handling home directory
+// expansion (~ or ~user) and environment variable expansion. It aims to be cross-platform
+func expandPath(path string) (string, error) {
+	if path == "" {
+		return "", nil
+	}
+
+	// the ~ expansion is a unix convention
```

Comment:
`expandHomeDir` already exists - we should make that a public function and use that instead.

Reply:
Sure 👍